Sources/Classes/RSEventRepository.m

```diff
@@ -167,13 +167,22 @@ - (void) __initiateSDK {
                         [RSLogger logDebug:@"EventRepository: Initiating ConsentFilterHandler"];
                         strongSelf->consentFilterHandler = [RSConsentFilterHandler initiate:strongSelf->config.consentFilter withServerConfig:serverConfig];
                     }
+                    
+                    // Check if SDK got updated from post DMT released version to >= 1.15.2, if it is then remove the previous unprocessed device mode events
+                    if ([self isUserHasUpdatedFromPostDMTReleaseVersion]) {
+                        [strongSelf->dbpersistenceManager updateEventsStatusToDeviceModeProcessingDone:[self->deviceModeManager getFirstEventTimestampBeforeSDKInit]];
+                        [strongSelf->dbpersistenceManager clearProcessedEventsFromDB];
+                    }
+                    
                     // initiate the native SDK factories if destinations are present
                     if (serverConfig.destinations != nil && serverConfig.destinations.count > 0) {
                         NSArray <RSServerDestination *> *consentedDestinations = self->consentFilterHandler != nil ? [self->consentFilterHandler filterDestinationList:serverConfig.destinations] : serverConfig.destinations;
                         if(consentedDestinations != nil && consentedDestinations.count > 0 ) {
                             [self->deviceModeManager startDeviceModeProcessor:consentedDestinations andDestinationsWithTransformationsEnabled:[strongSelf->configManager getDestinationsWithTransformationsEnabled]];
                         }
                     } else {
+                        [self->deviceModeManager sendPreviousUnprocessedDeviceModeEvents];
```

Comment:
Actually, if we are in the `else` branch it means that there are no device modes, can you please explain why are calling `sendPreviousUnprocessedDeviceModeEvents` in this case?

Reply:
If someone makes the event, even before SDK gets initialised is stored in the DB with status code 0, right?
We need to process that event once the SDK is initialised, hence, I'm calling `sendPreviousUnprocessedDeviceModeEvents` in the else block to mark the status of events as `device_mode_processing_done` and also to set [this](https://github.com/rudderlabs/rudder-sdk-ios/blob/510c8db7c360a428b8be8567020eb81663badc96/Sources/Classes/RSDeviceModeManager.m#L34) flag to `true`, so that post-SDK init any event sent will be marked as device mode processing done.

Reply:
Based on the discussion, I agree to refactor this logic and we basically decided not to send any events to makeFactoryDump in case no device mode factories are present, instead directly marking them as device_mode_processing_done.